IdentityCore/tests/util/cache/MSIDTestCacheFormat.m

```diff
@@ -0,0 +1,295 @@
+}
+
+
+- (MSIDToken *)getSharedRTForAccount:(MSIDAccount *)account
+                       requestParams:(MSIDRequestParameters *)parameters
+                             context:(id<MSIDRequestContext>)context
+                               error:(NSError **)error
+{
+    return [self getTokenForAccount:account
+                          tokenType:MSIDTokenTypeRefreshToken
+                             params:parameters
+                            context:context
+                              error:error];
+}
+
+- (NSArray<MSIDToken *> *)getAllSharedRTsWithParams:(MSIDRequestParameters *)parameters
+                                            context:(id<MSIDRequestContext>)context
+                                              error:(NSError **)error
+{
+    if (!parameters)
+    {
+        if (error)
+        {
+            *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, @"Missing parameter", nil, nil, nil, nil, nil);
+        }
+        
+        return nil;
+    }
+    
+    NSMutableArray *resultTokens = [NSMutableArray array];
+    
+    // Filter out tokens based on the clientId
+    for (NSString *key in [_cacheContents allKeys])
+    {
+        NSArray *contents = _cacheContents[key];
+        
+        for (MSIDToken *token in contents)
+        {
+            if (token.tokenType == MSIDTokenTypeRefreshToken
+                && [token.clientId isEqualToString:parameters.clientId])
+            {
+                [resultTokens addObject:token];
+            }
+        }
+    }
+    
+    return  resultTokens;
+}
+
+- (BOOL)removeSharedRTForAccount:(MSIDAccount *)account
+                           token:(MSIDToken *)token
+                         context:(id<MSIDRequestContext>)context
+                           error:(NSError **)error
+{
+    if (!account
+        || !token)
+    {
+        if (error)
+        {
+            *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, @"Missing parameter", nil, nil, nil, nil, nil);
+        }
+        
+        return NO;
+    }
+    
+    NSString *tokenIdentifier = [self tokenIdentifierForAccount:account tokenType:token.tokenType];
+    
+    NSMutableArray *accountTokens = _cacheContents[tokenIdentifier];
+    
+    if (accountTokens)
+    {
+        [accountTokens removeObject:token];
+    }
+    
+    return YES;
+}
+
+#pragma mark - Helpers
+
+- (BOOL)saveTokenForAccount:(MSIDAccount *)account
+                      token:(MSIDToken *)token
+                    context:(id<MSIDRequestContext>)context
+                      error:(NSError **)error
+{
+    if (!token
+        || !account)
+    {
+        if (error)
+        {
+            *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, @"Missing parameter", nil, nil, nil, nil, nil);
+        }
+        
+        return NO;
+    }
+    
+    NSString *tokenIdentifier = [self tokenIdentifierForAccount:account tokenType:token.tokenType];
+    
+    @synchronized (self) {
```

Comment:
nit: need this be a synchronized? are we going to run multiple thread testing off of this code?